boards/arm/weact_stm32g431_core/doc/index.rst

```diff
@@ -0,0 +1,155 @@
+.. _weact_stm32g431_core:
+
+WeAct Studio STM32G431 Core Board
+#################################
+
+The WeAct STM32G431 Core Board is a low-cost bare-bones
+STM32G431-based development board. See the `STM32G431CB website`_ for
+more information about the MCU. More information about the board,
+including schematics, is available from the `WeAct GitHub`_.
+
+Modifications USB-C Power Delivery
+**********************************
+
+The board does not support USB-C PD in its standard configuration. To
+enable USB-C PD, CC1 and CC2 need to be disconnected from their
+pull-down resistors and be connected to PB6 and PB4 respectively Dead
+battery support requires PA9 and PA10 to be routed to CC1 and
+CC2. VBUS also needs to be connected to the MCU through a voltage
+divider.
+
+The pull-downs are disconnected by removing the zero-Ohm resistors on
+SB8 and SB9 next to the USB-C connector. SB3, SB5, SB6, and SB7 then
+need to be closed to connect the CCx lines to the MCU. The voltage
+divider is connected to PB2 by closing SB4.
+
+After these modifications have been made, PA9, PA10, PB2, PB4, and PB6
+should be considered reserved for USB-C and not available for other
+applications.
+
+*Warning:* The internal USB DFU boot loader may not work correctly
+with machines that respect USB PD signaling after these modifications
+since it does not seem to configure the CCx lines correctly. A USB-C
+to USB-A adapter can be used or a separate SWD adapter can be used as
+a workaround.
```

Comment:
```
.. warning::
   The internal USB DFU boot loader may not work correctly with machines that 
   respect USB PD signaling after these modifications since it does not seem to
   configure the CCx lines correctly. A USB-C to USB-A adapter can be used or a
   separate SWD adapter can be used as a workaround.
```